src/components/PokemonTable/PokemonTable.js

```diff
@@ -0,0 +1,135 @@
+import React, {useEffect, useState} from "react";
+import {Link} from "react-router-dom";
+import {DataGrid} from '@material-ui/data-grid';
+import Container from "@material-ui/core/Container";
+import pokemonTableStyles from "./PokemonTable.style";
+import {pokemonService} from "../../service/PokemonService";
+import {HEADER_CELLS, POKEMON_SPRITE} from "../../constants";
+import TypeSelect from "../Select/Select.component";
+import SearchBox from "../SearchBox/SearchBox.component";
+import {debounce} from "../../utils";
+import {CustomNoRowsOverlayComponent} from "../CustomNoRowsOverlay/CustomNoRowsOverlay.component";
+
+export default function PokemonTable() {
+    const [pokemon, setPokemon] = useState([]);
+    const [filteredPokemon, setFilteredPokemon] = useState([]);
+    const [pokemonTypes, setPokemonTypes] = useState([]);
+    const [selectedPokemonTypes, setSelectedPokemonTypes] = useState([]);
+    const [searchField, setSearchField] = useState('');
+    const [pageSize, setPageSize] = useState(10);
+
+    const classes = pokemonTableStyles();
+
+    const uploadInitialPage = () => {
+        pokemonService.getPokemonTypes().then(types => {
+            setPokemonTypes(types.data);
+        });
+        pokemonService.getAllPokemon().then(pokemon => {
+            setPokemon(pokemon.data);
+            setFilteredPokemon(pokemon.data);
+        });
+    }
+
+    const onPokemonTypeSelect = (types) => {
+        setSelectedPokemonTypes(types);
+    }
+
+    const handleSearchRequest = (event) => {
+        setSearchField(event.target.value);
+    }
+
+    useEffect(() => {
+        if (pokemon && pokemon.length === 0) {
+            uploadInitialPage();
+        }
+    }, [pokemon]);
+
+    useEffect(() => {
+        let filteredPokemon = pokemon;
+        if (searchField) {
+            filteredPokemon = filteredPokemon.filter(pokemon =>
+                pokemon.name.toLowerCase().includes(searchField.toLowerCase())
+            )
+        }
+        if (selectedPokemonTypes && selectedPokemonTypes.length) {
+            const isTypeSelected = (type) => selectedPokemonTypes.includes(type);
+            filteredPokemon = filteredPokemon.filter(pokemon => pokemon.types.map(type => type.type.name).some(isTypeSelected));
+        }
+        setFilteredPokemon(filteredPokemon);
+    }, [pokemon, searchField, selectedPokemonTypes])
+
+    const columns = [
+        {
+            field: 'id',
+            headerName: HEADER_CELLS[0],
+            renderCell: params => (<Link to={'/pokemon/' + params.value}>{`# ${params.value}`}</Link>),
```

Comment:
I think it would be better to use history.push to redirect to pokemon page and create onClick handler for row